lib/ziti_utils.c

```diff
@@ -0,0 +1,207 @@
+/*
+Copyright 2019-2020 NetFoundry, Inc.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+https://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+#include <mbedtls/error.h>
+// #include <mjson.h>
+#include <uv.h>
+#include <uv_mbed/uv_mbed.h>
+#include "ziti_utils.h"
+
+
+#if !defined(ZITI_VERSION)
+#define ZITI_VERSION unknown
+#endif
+
+#if !defined(ZITI_BUILDNUM)
+#define ZITI_BUILDNUM <local>
+#endif
+
+#if !defined(ZITI_BRANCH)
+#define ZITI_BRANCH "<no-branch>"
+#define ZITI_COMMIT "<sha>"
+#endif
+
+#define to_str(x) str(x)
+#define str(x) #x
+
+/*
+ * https://sourceforge.net/p/predef/wiki/OperatingSystems/
+ */
+#if defined(WIN32)
+#define ZITI_OS Windows
+#elif defined(__ANDROID__)
+#define ZITI_OS Android
+#elif defined(__linux__)
+#define ZITI_OS Linux
+#elif defined(__APPLE__)
+#define ZITI_OS MacOS
```

Comment:
iOS eventually?

Reply:
Not sure. This file was copied from the ziti SDK for the logger. We're going to surface it in the ziti SDK though.